### ISSUE TITLE
idea: create tx/psbt in browser with mempool

### ISSUE THREAD

**Issue by antonilol** [mempool/mempool#1197]
### Description

a page on mempool where you can
- make a new transaction from a utxo or address
- export/download it as a `.psbt` file (see [here](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki) for exact specs) (`bitcoinjs-lib` can do this)
- maybe even sign it in the browser (with deterministic ecdsa, don't let a browser generate random numbers, `bitcoinjs-lib` can sign, i don't know if it supports deterministic ecdsa)

i would make it look like the transaction view (`/tx/:txid`) but then with input boxes/dropdowns/checkboxes etc...

#### Additional info

inspired by #178

**Comment by wiz** [comment]
Yeah this is what I had in mind with #178 - maybe @junderw wants to implement it :)

**Comment by junderw** [comment]
This is MIT licensed software made by bitcoinjs's new member @motorina0 which performs exactly what you want.

Try it out here:
https://bitcoincore.tech/apps/bitcoinjs-ui/index.html

Source:
https://github.com/bitcoincoretech/bitcoinjs-ui

idk about loading private keys into memory on a production app though.

**Comment by junderw** [comment]
I think a mempool version of this might have a way to load an xpub or address, click a button to list UTXOs, and click a button to automatically add them to the inputs of the PSBT editor.

**Comment by junderw** [comment]
TBH, the hardest part about this feature would be the UX. The bitcoinjs code is straightforward, especially if you map the UI 1:1 to every feature of our API (like the site above)

**Comment by junderw** [comment]
(hint: I suck at UI coding and to me 1:1 UX to the API is "easy to use" for me, since I made the API lol)

**Comment by wiz** [comment]
Cool. In the past I used coinb.in which had similar functionality to derive keys and addresses from xpubs and create custom multisig transactions etc. - would be cool to build it all into mempool UI so people could easily self host it on their own node

**Comment by junderw** [comment]
at that point... might as well just build a wallet. lol

coinbin is shitty UX and way too prone to foot cannons.

I actually worked on coinbin quite a bit before bitcoinjs.

looking back on my code, I cringe.
